scrapy_autoextract/providers.py

```diff
@@ -45,17 +51,88 @@ def __init__(
         self.stats = stats
         self.settings = settings
 
+    def __before__(self):
+        self.providers.append(self)
+
     async def __call__(self):
         """Make an AutoExtract request and build a Page Input of provided class
         based on API response data.
         """
-        page_type = self.get_page_type()
-        self.stats.inc_value(f"autoextract/{page_type}/total")
+        try:
+            if self.providers[-1] is self:
+                await self._fetch_data()
+        except IndexError:
+            # FIXME: it's not possible to use this alone yet
+            raise Exception("AutoExtractHTMLData should not be used alone")
```

Comment:
Today, if this list is empty it's because the object being queried is the AutoExtract HTML dependency. Ideally, we should solve this by:
- registering this provider in the `self.providers` list (its hook is being overridden without calling super)
- creating a new page type on AutoExtract backend (`webPage`, maybe?)

Reply:
Alternatively, we could arrange a way to fetch a dummy page type if the list is empty/we're only asking for AutoExtract HTML data.